aten/src/ATen/native/cuda/DilatedMaxPool2d.cu

```diff
@@ -129,6 +210,100 @@ __global__ void MaxPoolBackward(const int nthreads, const scalar_t* top_diff,
   }
 }
 
+template <typename scalar_t, typename accscalar_t>
+C10_LAUNCH_BOUNDS_1(CUDA_MAX_THREADS)
+__global__ void MaxPoolBackwardNHWC(const int nthreads, const scalar_t* top_diff,
+                                    const int64_t* top_mask, const int num, const int channels,
+                                    const int height, const int width, const int pooled_height,
+                                    const int pooled_width, const int kernel_h, const int kernel_w,
+                                    const int stride_h, const int stride_w, const int pad_h, const int pad_w,
+                                    const int dilation_h, const int dilation_w,
+                                    const int out_stride_c, const int out_stride_h, const int out_stride_w,
+                                    const int in_stride_c, const int in_stride_h, const int in_stride_w,
+                                    scalar_t* bottom_diff) {
+  extern __shared__ int smem[];
+  scalar_t *out_cached = reinterpret_cast<scalar_t*>(smem);
+
+  int thread_id = threadIdx.x + blockDim.x * (threadIdx.y + blockDim.y * threadIdx.z);
+  int block_size = blockDim.x * blockDim.y * blockDim.z;
+
+  for (int i = thread_id; i < channels*blockDim.y*blockDim.z; i+= block_size) {
+    out_cached[i] = scalar_t(0.0);
+  }
+  __syncthreads();
+  out_cached = &out_cached[(threadIdx.z * blockDim.y + threadIdx.y) * channels];
+
+  bottom_diff = bottom_diff + blockIdx.x * height * width * channels;
+  top_mask = top_mask + blockIdx.x * pooled_height * pooled_width * channels;
+  top_diff = top_diff + blockIdx.x * pooled_height * pooled_width * channels;
+
+  int iH = (height + gridDim.z-1) / gridDim.z;
+  int iW = (width + gridDim.y-1) / gridDim.y;
+  int istartH = threadIdx.z + blockIdx.z*iH;
+  int iendH = ::min(istartH+iH, height);
+  int istartW = threadIdx.y + blockIdx.y*iW;
+  int iendW = ::min(istartW+iW, width);
+
+  for (int ih = istartH; ih < iendH; ih+=blockDim.z) {
+    for (int iw = istartW; iw < iendW; iw+=blockDim.y) {
+      int phstart, phend, pwstart, pwend;
+      if (stride_h == 1) {
```

Comment:
NIT: Looks like we are using the same logic as with the NCHW kernel here. Maybe we want to combine the code and avoid two copies of the same code here. So later if we want to clean up the code, it would be easier.

Reply:
Fixed